### PR TITLE
Adding parameterized test for mii end-to-end use case in JUnit 5

### PR DESCRIPTION
**Pull request by vanajamukkara**
- Adding parameterized test for mii end-to-end use case in JUnit 5, WebLogic image tags can be passed as env var.
- Modified ItMiiDomain.java to call the utility methods in CommonTestUtils.java and removed duplicate code
- Deleted testCreateMiiDomainSameDomainUidDiffNS as parameterized test uses the same domainuid with different namespaces.

Jenkins results, clean run on Kind cluster - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/138/

on regular cluster, all tests ran fine except domain on pv test is failing with NPE, Sankar is working on a fix - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests-10/140/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 1]
```diff
@@ -259,60 +171,60 @@ public void testCreateMiiDomain() {
...
         adminServerPodName, domainNamespace);
-    checkPodCreated(adminServerPodName, domainUid, domainNamespace);
+    checkPodExists(adminServerPodName, domainUid, domainNamespace);
 
     // check managed server pods exist
     for (int i = 1; i <= replicaCount; i++) {
       logger.info("Check for managed server pod {0} existence in namespace {1}",
           managedServerPrefix + i, domainNamespace);
-      checkPodCreated(managedServerPrefix + i, domainUid, domainNamespace);
+      checkPodExists(managedServerPrefix + i, domainUid, domainNamespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:220]: You don't really need this check, the podReady actually validates the existence of the pod and ready status

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 2]
```diff
@@ -748,6 +573,117 @@ public void tearDownAll() {
     }
   }
 
+  @ParameterizedTest
+  @DisplayName("Create model in image domain using different WebLogic version images as parameters")
+  @MethodSource("oracle.weblogic.kubernetes.utils.Params#webLogicImageTags")
+  public void testParamsCreateMiiDomain(String imageTag, @Namespaces(1) List<String> namespaces) {
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:579]: add javadoc

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 3]
```diff
@@ -748,6 +573,117 @@ public void tearDownAll() {
...
+    logger.info("Check for admin server pod {0} existence in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodExists(adminServerPodName, domainUid, domainNamespace);
+
+    // check managed server pods exist
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check for managed server pod {0} existence in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:647]: we don't need the podExists assertions

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
   public static final String ADMIN_SERVER_NAME_BASE = "admin-server";
   public static final String MANAGED_SERVER_NAME_BASE = "managed-server";
   public static final String WLS_DOMAIN_TYPE = "WLS";
+  public static final String DEFAULT_WLS_IMAGE_TAGS = "12.2.1.3, 12.2.1.4";
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java:21]: Not 14.1.1.0? I think we should add that version.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 1 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 2 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:561]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 3 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java (hunk 4 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java]: modified

**Pull request merged**